homeassistant/components/freedns.py

```diff
@@ -26,21 +26,32 @@
 UPDATE_URL = 'https://freedns.afraid.org/dynamic/update.php'
 
 CONFIG_SCHEMA = vol.Schema({
-    DOMAIN: vol.Schema({
-        vol.Exclusive(CONF_URL, DOMAIN): cv.string,
-        vol.Exclusive(CONF_ACCESS_TOKEN, DOMAIN): cv.string,
-        vol.Optional(CONF_UPDATE_INTERVAL, default=DEFAULT_INTERVAL): vol.All(
-            cv.time_period, cv.positive_timedelta),
-
-    })
+    DOMAIN: vol.All(
+        vol.Schema({
+            vol.Exclusive(CONF_URL, DOMAIN): cv.string,
+            vol.Exclusive(CONF_ACCESS_TOKEN, DOMAIN): cv.string,
+            vol.Optional(CONF_UPDATE_INTERVAL):
+                vol.All(cv.time_period, cv.positive_timedelta),
+            vol.Optional(CONF_SCAN_INTERVAL):
+                vol.All(cv.time_period, cv.positive_timedelta),
+        }),
+        cv.deprecated(
+            CONF_UPDATE_INTERVAL,
+            replacement_key=CONF_SCAN_INTERVAL,
+            invalidation_version='1.0.0',
+            default=DEFAULT_INTERVAL
+        ),
+        cv.has_at_most_one_key(CONF_SCAN_INTERVAL, CONF_UPDATE_INTERVAL)
```

Comment:
Can we build this check into `cv.deprecated`?

Reply:
We can certainly call it from there. I thought it might potentially be useful outside of this use case as well so thought exposing it as a separate cv function was a good approach.

One potential use case I can think of is if we had a component that allowed login via email/password and an OAuth token. We wouldn't need all of those, so we could validate that only was provided.

Reply:
It's good to make it a separate function, but I think we shouldn't need to call it separately when using `cv.deprecated`. So I'm thinking we could use `cv.has_at_most_one_key` inside `cv.deprecated`.